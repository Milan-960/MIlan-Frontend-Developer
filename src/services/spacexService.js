//* this will call a noraml api
// const apiUrl = process.env.REACT_APP_API_BASE_URL;

//* this will call a PHP api
const apiUrl = "http://localhost:8000/api.php";

export async function fetchCapsules() {
  const response = await fetch(apiUrl);
  if (!response.ok) {
    throw new Error("API request failed");
  }
  const data = await response.json();
  return data;
}
