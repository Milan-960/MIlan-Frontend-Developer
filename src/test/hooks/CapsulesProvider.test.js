import React from "react";
import renderer from "react-test-renderer";

import { fetchCapsules } from "../../services/spacexService";

import {
  CapsulesProvider,
  CapsulesContext,
} from "../../hooks/CapsulesProvider";

jest.mock("../../services/spacexService");

// Dummy component that will consume the context
const Consumer = () => {
  const contextValue = React.useContext(CapsulesContext);
  return <div>{JSON.stringify(contextValue)}</div>;
};

test("CapsulesProvider provides correct context value", async () => {
  const mockData = [
    { type: "type1", status: "status1", original_launch: "launch1" },
  ];
  fetchCapsules.mockResolvedValue(mockData);

  const component = renderer.create(
    <CapsulesProvider>
      <Consumer />
    </CapsulesProvider>
  );

  // Let's give some time for the async function to resolve
  await renderer.act(() => new Promise((resolve) => setTimeout(resolve, 100)));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
