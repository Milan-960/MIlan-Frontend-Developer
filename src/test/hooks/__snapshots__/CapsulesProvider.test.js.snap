// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CapsulesProvider provides correct context value 1`] = `
<div>
  {"capsules":[{"type":"type1","status":"status1","original_launch":"launch1"}],"loading":false,"types":["type1"],"statuses":["status1"],"launches":["launch1"]}
</div>
`;
